sys/mips/mips/locore.S

```diff
@@ -163,9 +163,18 @@ VECTOR(_locore, unknown)
 	/*
 	 * Enable userspace cycle-count, cycle-count resolution, and CPU
 	 * number registers available via rdhwr.
+	 * Also enable beri specific statcounters.
 	 */
 	MFC0	t0, MIPS_COP_0_HWRENA
-	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4)
+	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4) \
+	| MIPS_BERI_HWRENA_RESET_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_ICACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_DCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2CACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_MEM_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2MASTER_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGMASTER_STATCOUNTERS
```

Comment:
I believe this change is in a `#if defined(CPU_BERI)`, so I don't know that a mask is necessary to fix it, but it would certainly simplify this code, so I will do it. The one in `sys/mips/beri/beri_mp.c` is not protected, but the file is in a `beri` directory, so I assume this is ok (even though I can see some `#if defined(CPU_CHERI)` else where in the file, which I am not sure what purpose they serve...) ? I want to define the mask to simplify the code as you suggested, but not define a `...BERI...` macro set to `0` in the non beri case. How does that sound ?